### ISSUE TITLE
Testing: positional piping when using the pipe operator with labelled function arguments

### ISSUE THREAD

**Issue by TanklesXL** [gleam-lang/gleam#1097]
Currently when the pipe operator `|>` is used to pipe into functions that have labelled arguments, it is possible to pipe to any position without the need for `_`

For example:
```rust
import gleam/should
import gleam/string

pub fn replace_test() {
  "Hello World!"
  |> string.replace(each: "l", with: "L")
  |> should.equal("HeLLo WorLd!")

  "Hello World!"
  |> string.replace(with: "L", each: "l")
  |> should.equal("HeLLo WorLd!")

  "l"
  |> string.replace(in: "Hello World!", with: "L")
  |> should.equal("HeLLo WorLd!")

  "l"
  |> string.replace(with: "L", in: "Hello World!")
  |> should.equal("HeLLo WorLd!")

  "L"
  |> string.replace(in: "Hello World!", each: "l")
  |> should.equal("HeLLo WorLd!")

  "L"
  |> string.replace(each: "l", in: "Hello World!")
  |> should.equal("HeLLo WorLd!")
}

```

Additional testing should be added to ensure that this behaviour is not broken with future compiler changes.

**Comment by lpil** [comment]
Yes let's make sure this preserved. Thank you!

**Comment by TanklesXL** [comment]
Btw @lpil I'll likely want to look at this in a few days, but I'm not very familiar with rust or with how the `gleam` repo is set up, what bits should I start looking at to familiarize myself & where do you think the testing would be needed, in the compiler itself or in some language/integration tests?

**Comment by lpil** [comment]
Heya! I think we could have tests in these two places

Integration tests: https://github.com/gleam-lang/gleam/blob/main/test/language/src/main.gleam

Typer tests: https://github.com/gleam-lang/gleam/blob/730f2ae9d1addfb7356f77e5ce8a5caa76ea0c05/src/type_/tests.rs#L3814-L3824

Let me know if those examples are unclear

**Comment by J3RN** [comment]
@lpil Is this solved with 0633622, or are further tests required?

**Comment by lpil** [comment]
Ah yes, this is done. Thank you
